spec.bs

```diff
@@ -928,6 +929,12 @@ On the other hand, methods for getting data from the [=shared storage database=]
     1. If |worklet|'s [=global scopes|list of global scopes=] is [=list/empty=], then return a [=promise rejected=] with a {{TypeError}}.
     1. [=Assert=] that |worklet|'s [=global scopes|list of global scopes=] [=list/contains=] a single {{SharedStorageWorkletGlobalScope}}.
     1. If the result of running [=SharedStorageWorkletGlobalScope/check whether addModule is finished=] for |worklet|'s {{SharedStorageWorkletGlobalScope}} is false, return a [=promise rejected=]  with a {{TypeError}}.
+    1. Optionally, return a [=promise rejected=] with a {{TypeError}}.
```

Comment:
@xyaoinum @pythagoraskitty I wasn't sure if this is the right place for returning an error from selectUrl and the right error type, can you help confirm if this aligns with the implementation or should this return be at an earlier point like in addModule, thanks!

Reply:
There seems to be two issues:
1. In this spec, we only check the user settings during storage getters/setters, but not for addModule()/selectURL()/run(). We should check the user settings for those as well.
2. In the spec, the Error is always "TypeError". In the code, the error type is sometimes [OperationError](https://webidl.spec.whatwg.org/#operationerror).

I can send a patch to fix (1) to unblock this patch (or @shivanigithub  do you prefer doing the fix along with this patch?). And problem 2 can be addressed separately as it's not only relevant to user setting errors.